warp/backend.go

```diff
@@ -30,14 +30,24 @@ type BlockClient interface {
 	GetAcceptedBlock(ctx context.Context, blockID ids.ID) (snowman.Block, error)
 }
 
+type MessageValidator interface {
+	// If the validator returns nil, the message is considered valid and the
+	// backend will sign it.
+	ValidateMessage(*avalancheWarp.UnsignedMessage) error
+}
+
 // Backend tracks signature-eligible warp messages and provides an interface to fetch them.
 // The backend is also used to query for warp message signatures by the signature request handler.
 type Backend interface {
 	// AddMessage signs [unsignedMessage] and adds it to the warp backend database
 	AddMessage(unsignedMessage *avalancheWarp.UnsignedMessage) error
 
-	// GetMessageSignature returns the signature of the requested message hash.
-	GetMessageSignature(messageID ids.ID) ([bls.SignatureLen]byte, error)
+	// AddMessageValidator adds a validator to the backend. The backend will sign
+	// messages that pass any of the validators, in addition to those known in the db.
```

Comment:
IMHO it's kind-of odd to expect message to be validated if "any" of these validators pass. Maybe I did not fully understand the case but shouldn't we just need to have validators on payload types rather than adding them to here?